src/tss2-esys/api/Esys_StartAuthSession.c

```diff
@@ -469,9 +469,20 @@ Esys_StartAuthSession_Finish(
                 LOG_ERROR("Out of memory.");
                 return TSS2_ESYS_RC_MEMORY;
             }
-            if  (bind != ESYS_TR_NONE && bindNode != NULL
-                 && bindNode->auth.size > 0)
-                memcpy(&secret[0], &bindNode->auth.buffer[0], bindNode->auth.size);
+
+            if  (bind != ESYS_TR_NONE && bindNode != NULL) {
+                /*
+                 * TPM2.0 Architecture 19.6.5 Note 2
+                 *
+                 * Remove tailing zeroes from the auth value
+                 */
+                while ((bindNode->auth.size > 0) &&
```

Comment:
I think the auth.size > 0 check needs to be part of the original if statement and then use a working variable for the removal of trailing zeroes. It is more steps, but safer in my opinion.

Reply:
@tomoveu You are right, the `bindNode->auth` shouldn't be changed. But since this is passed to `iesys_compute_bound_entity` later, the `TPM2B_DIGEST` structure needs to be copied into a working variable I name `bindNodeAuth`.

The computation of `secret_size` was off in the original merge request, which I corrected by moving the removal of trailing zeros.

Finally, some lines below can be simplied since `bindNodeAuth.size` is always valid and 0 if `bindNode == NULL`.